enterprise/ssl-impl/src/main/java/io/crate/protocols/ssl/FilesWatcher.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.protocols.ssl;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+
+/**
+ * The file watcher checks for changes on registered files
+ * and notifies their listeners accordingly.
+ */
+class FilesWatcher implements Runnable, Closeable {
+
+    private final WatchService watchService;
+    private final Map<WatchKey, List<Watcher>> watchers = new HashMap<>();
+
+    FilesWatcher() throws IOException {
+        this.watchService = FileSystems.getDefault().newWatchService();
+    }
+
+    void addListener(Path path,
+                     Consumer<WatchEvent<Path>> listener,
+                     WatchEvent.Kind<?>... watchEventKinds) throws IOException {
+        Watcher watcher = new Watcher(path, listener);
+        WatchKey key = watcher.registrablePath().register(watchService, watchEventKinds);
```

Comment:
The `Watchable` interface mentions that a watchKey should be cancelled to "de-register" - We should probably go over all keys in the `close` and cancel them?